### PR TITLE
add _android

### PR DESCRIPTION
**Pull request by axet**
Not every android has proper route table. But for some it works. Also I add ping discover. Not sure if it is good idea :) Not every routers ppp connections respond to ping -t 0.

### CODE REVIEW

File: gateway_android.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return parseLinuxRoute(output)
+}
+
+// unsucessful ping discover can take up to 1 sec
+func discoverGatewayUsingPing() (net.IP, error) {
+	routeCmd := exec.Command("/system/bin/ping", "-n", "-c", "1", "-t", "0", "-W", "1", "198.41.0.4") // a.root-servers.net
```
Comment (jackpal) [inline gateway_android.go]: Are you sure this should be "-t 0", and not "-t 1"?

If I try (on a random Linux box I happen to have handy, I get):

$ ping -n -c 1 -t 0  -W 1 198.41.0.4
ping: can't set unicast time-to-live: Invalid argument

But if I try with -t 1, it works

$ ping -n -c 1 -t 1  -W 1 198.41.0.4
PING 198.41.0.4 (198.41.0.4) 56(84) bytes of data.
From 192.168.1.1: icmp_seq=1 Time to live exceeded

--- 198.41.0.4 ping statistics ---
1 packets transmitted, 0 received, +1 errors, 100% packet loss, time 0ms

$

**Comment by jackpal** [comment]
Is there an Android device you know of, that doesn't have route or ip, but does have ping?

Also it looks like the commands to ping might not be correct -- see my comment in your CL.

**Comment by axet** [comment]
I'm not yet happy about a solution. On my android phone it works only when I use wifi connection and it stuck for a 1 second with no result while I'm on mobile data.

And yes I did test it booth -t 1 && -t 0 works on my phone and route table are empty (may be all devices with android 6.0 have no correct route table).

Google documentation says we should use RouteInfo... lets me dig for a day...

**Comment by axet** [comment]
Seems not worth it. I can't ensure gateway have been detected with any option:

- getprop
- ifconfig
- /etc
- route
- ip
- ping

And I do not like ping timeout issue. So. I'll close pull request, and keep changes in my tree. Feel free to merge manually.

**Comment by jackpal** [comment]
Thanks for trying!

**Pull request closed without merging**